spec/api_spec.rb

```diff
@@ -146,37 +146,137 @@
+
+        let(:old_email) { 'test+001@test.com' }
+        let(:new_email) { 'updated_email@example.com' }
+
+        before do
+          stub_find_maropost_contact(email: old_email, status: 200)
+          stub_find_maropost_contact(email: new_email, status: 404)
+          stub_do_not_mail_list_exists(email: old_email, status: 404)
+          stub_do_not_mail_list_exists(email: new_email, status: 404)
+          stub_update_maropost_contact(contact_id: 741_000_000, status: 200, body: email_updated_contact)
+          stub_do_not_mail_list_delete(email: new_email)
+        end
+
+        it 'returns updated contact' do
+          contact = subject
+
+          expect(contact.email).to eq new_email
+        end
       end
 
-      it 'returns updated contact' do
-        contact = subject
+      context 'maropost contact does not exist with old email' do
+        let(:old_email) { 'non_existent_user@example.com' }
+        let(:new_email) { 'new_non_existent_user@example.com' }
 
-        expect(contact.email).to eq new_email
+        before do
+          stub_find_maropost_contact(email: old_email, status: 404, body: '{"message": "Contact is not present!"}')
+          stub_find_maropost_contact(email: new_email, status: 404, body: '{"message": "Contact is not present!"}')
+        end
+
+        it 'returns nil' do
+          expect(subject).to be_nil
+        end
       end
     end
 
-    context 'contact does not exist' do
-      let(:old_email) { 'non_existent_user@example.com' }
-      let(:new_email) { 'new_non_existent_user@example.com' }
+    context 'contact exists in maropost with the new email' do
+      context 'maropost contact exists with old email' do
+        let(:old_contact_maropost_response) { read_fixture('contacts', 'contact.json') }
+        let(:existing_contact_maropost_response) { read_fixture('contacts', 'other_existing_contact.json') }
+        let(:merged_contact_maropost_response) { read_fixture('contacts', 'merged_contact.json') }
+
+        let(:old_email) { 'test+001@test.com' }
+        let(:new_email) { 'test@example.com' }
+
+        before do
+          stub_find_maropost_contact(email: old_email, status: 200, body: old_contact_maropost_response)
+          stub_find_maropost_contact(email: new_email, status: 200, body: existing_contact_maropost_response)
+          stub_do_not_mail_list_exists(email: old_email, status: 404)
+          stub_do_not_mail_list_exists(email: new_email, status: 404)
+
+          stub_update_maropost_contact(contact_id: 888_000_000, status: 200, body: merged_contact_maropost_response)
+          
+          stub_do_not_mail_list_delete(email: new_email)
+          stub_do_not_mail_list_create(email: old_email)
+        end
 
-      before do
-        stub_find_maropost_contact(email: old_email, status: 404, body: '{"message": "Contact is not present!"}')
+        it 'applies old maropost contact settings to existing maropost contact' do
+          updated_contact = subject
```

Comment:
why do you need this? can the calls to attributes below be made directly on subject?

Reply:
It would call subject for each thing I am testing.